tests/test_main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+from unittest import mock
+from app.main import cryptocurrency_action
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_rate_is_more(mocked_rate_prediction):
+    mocked_rate_prediction.return_value = 29.68
+
+    assert cryptocurrency_action(28) == "Buy more cryptocurrency"
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_rate_is_less(mocked_rate_prediction):
+    mocked_rate_prediction.return_value = 23.5
+
+    assert cryptocurrency_action(25) == "Sell all your cryptocurrency"
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_rate_is_slightly_higher(mocked_rate_prediction):
+    mocked_rate_prediction.return_value = 25.23
+
+    assert cryptocurrency_action(24.5) == "Do nothing"
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_rate_is_slightly_lower(mocked_rate_prediction):
+    mocked_rate_prediction.return_value = 23.34
+
+    assert cryptocurrency_action(23.82) == "Do nothing"
```

Comment:
Add boundary conditions, you don't check when `get_exchange_rate_prediction` return value equals exactly 1.05 or 0.95 * value passed in `cryptocurrency_action`.